spec/rspec/matchers/built_in/be_spec.rb

```diff
@@ -85,6 +85,21 @@
         expect(actual).to be_happy
       }.to fail_with("expected `#{actual.inspect}.happy?` to be truthy, got false")
     end
+
+    it "does not allow non-supplied dynamic matchers to pass on spies" do
+      thing = spy("thing", has_recv?: true)
+
+      expect(thing).to have_recv(:foo)
+      expect(thing).to have_received(:has_recv?).with(:foo)
```

Comment:
Can you please split this into two examples? Otherwise the one that covers the “really” part is easily missed.

Reply:
I was wavering on doing that in the first place, but without the fail-matcher the specs were so big and ugly that I didn't want to double them up. You guys were responding so fast, I figured it'd be better to just get something up and find out what you liked :-)